docs/design/provisioning.md

```diff
@@ -0,0 +1,143 @@
+
+`PersistentVolumeSelector` on PersistentVolumeClaim has been proposed as a means to select specific volumes, akin to a pod's NodeSelector.
+
+Having both fields is confusing.  Consider this on-premise installation of Kube:
+
+> Admin Joe configures "Gold" to dynamically provision Ceph volumes.  Joe also creates "Silver" volumes by manually provisioning a large number of NFS exports on expensive hardware (SSDs, highly available, stringent backup policy).  Lastly, Joe creates "Bronze" NFS exports on older hardware with slower discs.
+
+> How does a user request Gold/Silver/Bronze?   Using the StorageClass field for "Gold" while requiring the use of VolumeSelector for Silver/Bronze requires a user to know the difference between volumes.
+
+Using just VolumeSelector gives a user a single path to understand.  Admins retain the ability to manually create storage.  Labels on the volume (or the volume creator) are matched with the claim's volume selector.
+
+An additional benefit of using a selector for everything is the ability to deprecate PersistentVolumeAccessModes.  These fields on PV are descriptors only.  They have no functional value other than to describe the capabilities of the volume.  AccessModes on a PVC represent only a way to select certain volumes.  Admins can use labels to describe a volume's capabilities (e.g, ReadWriteMany) and users would select volumes accordingly.
+
+
+
+
+
+#### Other approaches considered:
+
+* Single ConfigMap with each key in Data representing 1 storage class, the value is opaque json.  Good for versioning the blob but it seemed unwieldy.
+* Many ConfigMaps with "storage-class" as a key in the Data.  Difficult to validate unique storage names.
+* Name of ConfigMap as name of storage class. Naming validation is enforced by API server.  Requires PVC.Spec.StorageClass (or similar) with 1:1 mapping of claim to provisioner.
+
+
+
+### Plugin Parameters
+
+The example above contains `"params": "${OPAQUE-JSON-MAP}"`, which is a string representing a JSON map of parameters.   Each provisionable volume plugin will document and publish the parameters it accepts.
+
+Plugin params can vary by whatever attributes are exposed by a storage provider.  An AWS EBS volume, for example, can be a general purpose SSD (gp2), provisioned IOPs (io1) for speed and throughput, and magnetic (standard) for low cost.  Additionally, EBS volumes can optionally be encrypted.  Configuring each volume type with optional encryption would require 6 distinct provisioners.  Administrators can mix and match any attributes exposed by the provisioning plugin to create distinct storage classes.
+
+
+## Controller behavior
+
+`PersistentVolumeBinderController` continues to watch claims and attempts to bind them to volumes, creating them as necessary/possible.
+
+The binder attempts to fulfill a claim in the following order:
+
+* If `pvc.Spec.PersistentVolumeSelector` is non-nil and matches a provisioner's labels, create a PV using the provisioner
```

Comment:
It could go either way and I have no preference.  

On one hand, we could check first just in case there's a matching volume and reduce inventory.  On the other hand, we could dynamically create an exact match (capacity-wise) for the claim.